extensions/opentelemetry/runtime/pom.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-opentelemetry-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-opentelemetry</artifactId>
+    <name>Quarkus - OpenTelemetry - Runtime</name>
+    <description>Use OpenTelemetry to trace services</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-vertx-http</artifactId>
```

Comment:
I would have expected this to be optional? Can't OpenTelemetry be used to measure all sorts of things?

Reply:
It can, but I should've clarified that this initial version is focused on HTTP tracing only.

There are follow on tasks for Reactive Messaging, REST Client, etc.

I was planning to over time add these bits once the initial extension is present. Is that a bad approach?

So in this case, once Reactive Messaging support is added it would be an optional dependency

Reply:
I don't think it's a bad approach. What I don't understand is why the Vert.x HTTP dependency is not optional right now? Even if we don't wire things for other components yet, people can use it for their own purposes without a need for Vert.x HTTP?

Reply:
As I implemented it with a set of assumptions, namely that Vert.x HTTP is always present.

I can look to fix that before the PR is no longer DRAFT, but it simplified the "getting it going" phase.

Reply:
@gsmet If I need to create a `FilterBuildItem`, how can I do that in a way that doesn't break if Vert.x HTTP isn't present? Is it adding an `onlyIf` check to the build step to verify it's there?

Reply:
Hum, you're right, that's going to be an issue.

Reply:
this might be a candidate for Alexey's new conditional extension approach.. though (as with micrometer) this will lead to a bunch of baby optional/hidden extensions ..

Reply:
It most definitely will be.

That's not merged yet though, if I recall?

If it is I can do it in a follow up PR

Reply:
Not merged yet. So definitely a follow-up/follow-on

Reply:
Make it optional.
Check the presence of a class, if not present, return null, otherwise, return your filter.
This is how I'll do it.
We use this scheme in opentracing, metrics, ...